### PR TITLE
doc/user: monitoring docs for Grafana

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

This PR implements the monitoring guide for Grafana. There is already one for Datadog, this is almost the same but for Grafana. The difference between this guide and the Datadog guide is that it applies for the Cloud and the Self-Managed version of Grafana.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by dehume — APPROVED** [review]

**Review by bobbyiliev — APPROVED** [review]

File: doc/user/content/manage/monitor/grafana.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+   ![Gif](https://github.com/MaterializeInc/demos/assets/11491779/e512a95f-e3c6-433d-bc8f-6f5138b08115)
+
+   </details>
+
+{{< /tab >}}
+{{< tab "Self-managed">}}
+1. Follow the [instructions to install and run Prometheus](https://prometheus.io/docs/prometheus/latest/installation/)
+   in your host.
+
+2. To configure a [Prometheus scrape](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config) edit the
```
Comment (morsapaes) [inline doc/user/content/manage/monitor/grafana.md:149]:
```suggestion
2. To configure a [Prometheus scrape](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config), edit the
```

File: doc/user/content/manage/monitor/grafana.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+         scrape_interval: 15s
+         static_configs:
+            - targets: ['sql-exporter:9237']
+            labels:
+               instance: sql_exporter
+   ```
+
+     **Tip:** see [this sample](https://github.com/MaterializeInc/demos/blob/main/integrations/grafana/local/prometheus.yml) for all available configuration options.
+
+3. Add **Prometheus** as a new source to Grafana.
```
Comment (morsapaes) [inline doc/user/content/manage/monitor/grafana.md:165]:
```suggestion
3. Follow [the instructions](https://grafana.com/docs/grafana/latest/datasources/prometheus/) to add **Prometheus** as a new data source in Grafana.
```

Comment (morsapaes) [reply doc/user/content/manage/monitor/grafana.md:165]: We have a sample data source file [here](https://github.com/MaterializeInc/demos/blob/main/integrations/grafana/local/misc/datasources/prometheus.yml) that we should mention?

File: doc/user/content/manage/monitor/grafana.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+
+For more details on how to configure, run and troubleshoot Prometheus, see the [Prometheus documentation](https://prometheus.io/docs/introduction/overview/).
+{{< /tab >}}
+{{< /tabs >}}
+
+## Step 3. Build a monitoring dashboard
+
+With the Prometheus SQL Exporter running SQL queries againt your Materialize
+region and exporting the results as metrics, and the Grafana Agent routing
```
Comment (morsapaes) [inline doc/user/content/manage/monitor/grafana.md:175]:
```suggestion
region and exporting the results as metrics, and a scraper routing
```

File: doc/user/content/manage/monitor/grafana.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+these metrics to your Grafana account, you're ready to build a monitoring
+dashboard!
+
+**Tip:** use [this sample](https://github.com/MaterializeInc/demos/blob/main/integrations/grafana/local/misc/dashboards/dashboard.json)
+to bootstrap a new dashboard with the key Materialize metrics and indicators
+defined in the sample `config.yml`.
+
+1. **Log in** to your Grafana account.
```
Comment (morsapaes) [inline doc/user/content/manage/monitor/grafana.md:183]: This is specific to Grafana Cloud, no?

**Review by morsapaes — CHANGES_REQUESTED** [review]

**Comment by morsapaes** [comment]
Ah, I missed the boat on the review; sorry. I can push the adjustments in as a separate PR. Thanks for working on this, @joacoc!

**Comment by joacoc** [comment]
@morsapaes I can do it now! Give me one min.

File: doc/user/content/manage/monitor/grafana.md (hunk 2 above)
Comment (joacoc) [reply doc/user/content/manage/monitor/grafana.md:165]: Yes!

**Pull request merged**